### PR TITLE
Add related packages. 

### PR DESCRIPTION
**Pull request by yebai**
I added a few related Julia libraries for probabilistic programming and more broadly Bayesian computation. Please feel free to suggest more related packages and/or edit the summary for packages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,5 +36,13 @@ Turing is an open source project so if you feel you have some relevant skills an
 
 Join [our channel](https://julialang.slack.com/messages/turing/) (`#turing`) in the Julia Slack chat for help, discussion, or general communication with the Turing team. If you do not already have an invitation to Julia's Slack, you can get one by going [here](https://slackinvite.julialang.org/).
 
+## Related packages
+
+- Probabilistic programming via source rewriting - [Soss.jl](https://github.com/cscherrer/Soss.jl)
+- A general-purpose probabilistic programming system with programmable inference - [Gen.jl](https://probcomp.github.io/Gen/)
+- Bare-bones implementation of robust dynamic Hamiltonian Monte Carlo methods - [DynamicHMC.jl](https://github.com/tpapp/DynamicHMC.jl)
+- Comparing performance and results of mcmc options using Julia - [MCMCBenchmarks.jl](https://github.com/StatisticalRethinkingJulia/MCMCBenchmarks.jl)
+
```
Comment (cpfiffer) [inline README.md]:
```suggestion
- [Omega.jl](https://github.com/zenna/Omega.jl) for probabilistic programming and causal inference.
```

**Comment by cpfiffer** [comment]
Should we add CmdStan?

**Comment by yebai** [comment]
> Should we add CmdStan?

Sounds good to me!

**Comment by trappmartin** [comment]
Maybe we should add a "Turing compared to ... " section, similar to the one used in [Omega](http://www.zenna.org/Omega.jl/latest/omegavsotherppls/#Omega-vs-other-Probabilistic-Programming-Languages-1).

**Pull request merged**